contracts/src/access/ownable/mod.rs

```diff
@@ -57,25 +66,27 @@ impl Default for Data {
 #[modifier_definition]
 pub fn only_owner<T, F, R, E>(instance: &mut T, body: F) -> Result<R, E>
 where
-    T: Storage<Data>,
+    T: Storage<DataType> + StorageAccess<Data>,
     F: FnOnce(&mut T) -> Result<R, E>,
     E: From<OwnableError>,
 {
-    if instance.data().owner != T::env().caller() {
+    if instance.get_or_default().owner != T::env().caller() {
         return Err(From::from(OwnableError::CallerIsNotOwner))
     }
     body(instance)
 }
 
-pub trait OwnableImpl: Storage<Data> + Internal {
+pub trait OwnableImpl: Storage<DataType> + StorageAccess<Data> + Internal {
     fn owner(&self) -> AccountId {
-        self.data().owner.clone()
+        self.get_or_default().owner.clone()
     }
 
     #[modifiers(only_owner)]
     fn renounce_ownership(&mut self) -> Result<(), OwnableError> {
-        let old_owner = self.data().owner.clone();
-        self.data().owner = ZERO_ADDRESS.into();
+        let old_owner = self.get_or_default().owner.clone();
+        with_data!(self, data, {
+            data.owner = ZERO_ADDRESS.into();
+        });
         self._emit_ownership_transferred_event(Some(old_owner), None);
```

Comment:
it is better to perform such changes independently, because if we will perform cross-contract calls between two changes, then changes won't be applied before such cross-contract call